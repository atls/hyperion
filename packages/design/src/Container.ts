import styled                     from '@emotion/styled'
import { flexbox, layout, space } from 'styled-system'

export const Container = styled.div(layout, space, flexbox)

Container.defaultProps = {
  flexDirection: 'column',
  display: 'flex',
  mx: 10,
  my: 20,
}
