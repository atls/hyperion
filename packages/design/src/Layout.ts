import { flexbox, layout, space } from 'styled-system'

import styled                     from '@emotion/styled'

export const Column = styled.div(layout, space, flexbox)

Column.defaultProps = {
  flexDirection: 'column',
  display: 'flex',
}

export const Row = styled.div(layout, space, flexbox)

Row.defaultProps = {
  flexDirection: 'row',
  display: 'flex',
}

export const Layout = styled.div(layout, space, flexbox)

Layout.defaultProps = {
  display: 'flex',
}
