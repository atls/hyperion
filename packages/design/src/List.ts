import styled              from '@emotion/styled'
import { flexbox, layout } from 'styled-system'

export const List = styled.div(layout, flexbox)

List.defaultProps = {
  display: 'flex',
  flexWrap: 'wrap',
  justifyContent: 'space-between',
}
