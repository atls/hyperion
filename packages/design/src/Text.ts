import { color, space, typography } from 'styled-system'

import styled                       from '@emotion/styled'

export const Text = styled.span(color, space, typography)

Text.defaultProps = {
  fontFamily: 'primary',
  fontWeight: 'normal',
  fontSize: 'normal',
}
