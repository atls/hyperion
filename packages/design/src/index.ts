export * from './Sample'
export * from './Layout'
export * from './Container'
export * from './List'
export * from './Text'
export * from './Space'
