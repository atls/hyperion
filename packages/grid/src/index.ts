export * from './Column'
export * from './Grid'
