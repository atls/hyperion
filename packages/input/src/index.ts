export * from './Textarea'
export * from './Input'
export * from './Checkbox'
