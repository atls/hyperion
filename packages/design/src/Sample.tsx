import React                                from 'react'
import styled                               from '@emotion/styled'
import { border, boxShadow, color, layout } from 'styled-system'

import { Container }                        from './Container'
import { Text }                             from './Text'

const SampleBox = styled.div(layout, border, boxShadow, color)

SampleBox.defaultProps = {
  width: 130,
  height: 80,
}

export const Sample = ({ name = '', value = '', ...props }) => (
  <Container>
    <SampleBox {...props} />
    <Text mt='10px' fontWeight='medium'>
      {name}
    </Text>
    <Text mt='5px' color='stardust' fontSize='small'>
      {value}
    </Text>
  </Container>
)
