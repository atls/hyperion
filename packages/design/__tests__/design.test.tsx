import React                                                         from 'react'
import renderer                                                      from 'react-test-renderer'

import { Column, Container, Layout, List, Row, Sample, Space, Text } from '../src'

describe('test suit for design', () => {
  it('should render Sample correctly', () => {
    const tree = renderer.create(<Sample />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render Layout correctly', () => {
    const tree = renderer.create(<Layout />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render List correctly', () => {
    const tree = renderer.create(<List />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render Space correctly', () => {
    const tree = renderer.create(<Space />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render Text correctly', () => {
    const tree = renderer.create(<Text>Any text</Text>).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render Column correctly', () => {
    const tree = renderer.create(<Column>Inside column</Column>).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render Container correctly', () => {
    const tree = renderer.create(<Container>Contains</Container>).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render Row correctly', () => {
    const tree = renderer.create(<Row>Inside row</Row>).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
