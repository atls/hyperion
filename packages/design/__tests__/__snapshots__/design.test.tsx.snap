// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test suit for design should render Column correctly 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="emotion-0"
  display="flex"
>
  Inside column
</div>
`;

exports[`test suit for design should render Container correctly 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 10px;
  margin-right: 10px;
  margin-top: 20px;
  margin-bottom: 20px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="emotion-0"
  display="flex"
>
  Contains
</div>
`;

exports[`test suit for design should render Layout correctly 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="emotion-0"
  display="flex"
/>
`;

exports[`test suit for design should render List correctly 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

<div
  className="emotion-0"
  display="flex"
/>
`;

exports[`test suit for design should render Row correctly 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
  className="emotion-0"
  display="flex"
>
  Inside row
</div>
`;

exports[`test suit for design should render Sample correctly 1`] = `
.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 10px;
  margin-right: 10px;
  margin-top: 20px;
  margin-bottom: 20px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0 {
  width: 130px;
  height: 80px;
}

.emotion-1 {
  margin-top: 10px;
  font-weight: medium;
  font-family: primary;
  font-size: normal;
}

.emotion-2 {
  color: stardust;
  margin-top: 5px;
  font-size: small;
  font-family: primary;
  font-weight: normal;
}

<div
  className="emotion-3"
  display="flex"
>
  <div
    className="emotion-0"
    height={80}
    width={130}
  />
  <span
    className="emotion-1"
    fontFamily="primary"
    fontSize="normal"
    fontWeight="medium"
  >
    
  </span>
  <span
    className="emotion-2"
    color="stardust"
    fontFamily="primary"
    fontSize="small"
    fontWeight="normal"
  >
    
  </span>
</div>
`;

exports[`test suit for design should render Space correctly 1`] = `
<span
  style={
    Object {
      "display": "inline-flex",
    }
  }
>
  Â 
</span>
`;

exports[`test suit for design should render Text correctly 1`] = `
.emotion-0 {
  font-family: primary;
  font-weight: normal;
  font-size: normal;
}

<span
  className="emotion-0"
  fontFamily="primary"
  fontSize="normal"
  fontWeight="normal"
>
  Any text
</span>
`;
